### DISCUSSION TITLE
Can you continue tests sequentially from prior test state?

### DISCUSSION THREAD

**Discussion by 0xFloop** [foundry-rs/foundry#1755] [Help]
As the title states, I'm wondering if there is a way within Forge to continue your tests sequentially, starting from the end state of the last test without repasting the prior tests code as setup. Obviously I could just make one massive test, however, i would lose the gas data and such that i would receive from individual tests. Thank you in advance to anyone who can help :)

**Comment by onbjerg** [comment]
Not really possible, no. The current way to do something like this is to use deployment patterns like this: https://github.com/gakonst/v3-periphery-foundry/blob/main/contracts/foundry-tests/utils/Deploy.sol

Essentially, you have a base fixture that inherits from `Test` and provides a `setUp` function. You can then extend this fixture to add more on top, again extending `setUp` (remember to call `super.setUp()` in your test and any sub fixtures).

Anything state `setUp` is shared between all tests
